src/circllhist.c

```diff
@@ -601,9 +607,16 @@ hist_approx_quantile(const histogram_t *hist, const double *q_in, int nq, double
   int i_q, i_b;
   double total_cnt = 0.0, bucket_width = 0.0,
          bucket_left = 0.0, lower_cnt = 0.0, upper_cnt = 0.0;
-  ASSERT_GOOD_HIST(hist);
+
   if(nq < 1) return 0; /* nothing requested, easy to satisfy successfully */
 
+  if(!hist) {
```

Comment:
The comments say this returns -1 for an empty histogram, instead of returning all NaNs:
```
/* 0 success,
 * -1 (empty histogram),
 * -2 (out of order quantile request)
 * -3 (out of bound quantile)`
 */
```
Not sure- which behavior is best?

Reply:
I like nan, but I defer to Heinrich

Reply:
Quantiles should be supported for empty histograms. We have to deal with missing data and empty histograms a lot.

I think the following semantics are sensible:
- returning an error code: the computation should be aborted.
- returning NaNs: this method does not make sense for the data provided. Keep going.

So, I vote for NaNs.